### DISCUSSION TITLE
Is the aggregate function permutation aware ?

### DISCUSSION THREAD

**Discussion by divideconcept** [pyannote/pyannote-audio#1394] [Development]
Hi, looking at the [aggregate function](https://github.com/pyannote/pyannote-audio/blob/74939acbfa830521a434cb4068176196dd9612dc/pyannote/audio/core/inference.py#L448) (whose purpose if I understand well is to combine multiple inferences of a sliding window) it seems that it doesn't do any detection of potential permutations/swappings, as illustrated here: https://herve.niderb.fr/fastpages/2022/10/23/One-speaker-segmentation-model-to-rule-them-all.html#What--pyannote/segmentation-CANNOT-do ?

**Comment by hbredin** [comment]
You got it right: it does not handle permutations.
